.github/workflows/docker-publish.yml

```diff
@@ -0,0 +1,43 @@
+name: Docker Publish
+
+on:
```

Comment:
I would suggest we also have a dispatch trigger too

Reply:
@nully0x what do you envision that being used for?

Reply:
A dispatch helps to manually trigger action. That's useful when an action fails.

Reply:
AFAIK you can still rerun a failed action without a dispatch trigger.

Reply:
Yea I am aware we can rerun jobs.

For me dispatch helps with clean execution and also helps to distinguish events in the action workflow.